components/brave_wallet/browser/eth_data_parser.cc

```diff
@@ -341,6 +342,55 @@ GetTransactionInfoFromData(const std::vector<uint8_t>& data) {
                                                     "uint128",   // sell amount
                                                     "uint128"},  // buy amount
                            tx_args);
+  } else if (selector == kCowOrderSellEthSelector) {
```

Comment:
This function could use a bit of refactoring. It's starting to get pretty complex for a parsing function that's handling untrusted data supplied by a page. Given that an error in this logic can easily lead to a rule of 2 violation we should split this up better.

Let's do a follow up issue for this.